### PR TITLE
Product page coverted to react

### PR DESCRIPTION
**Pull request by surbhi84**
Switch from html to react

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+
+        <Route
+          path="/ts"
+          element={
+            <div style={{ margin: "100px" }}>
+              <Mockman />
+            </div>
+          }
+        />
+        {/*the above route is for testing purposes */}
```
Comment (rohanmathur91) [inline src/App.js]: Always have comment first then the code, ie put line 23 to line number 14

File: src/backend/db/products.js [hunk 2]
```diff
@@ -7,24 +7,110 @@ import { v4 as uuid } from "uuid";
...
+  //   product: ,
+  //   productImg: ,
+  //   productAlt: ,
+  //   discPrice: 1349,
+  //   price: 2139,
+  //   discount: 35,
+  //   inStock: true,
+  //   rating: 4,
+  //   quantity: 0,
+  // },
```
Comment (rohanmathur91) [inline src/backend/db/products.js:115]: Remove commented code

File: src/backend/db/users.js [hunk 3]
```diff
@@ -17,3 +17,8 @@ export const users = [
     updatedAt: formatDate(),
   },
 ];
+
+/* 
+"email": "adarshbalika@gmail.com",
+"password": "adarshbalika"
+*/
```
Comment (rohanmathur91) [inline src/backend/db/users.js:24]: same as above

File: src/components/carousel/Carousel.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+              setActiveSlide(item);
+            }}
+            className={`${carousel["manual-btn"]}  
+              ${item === activeSlide ? carousel["active"] : "abcd"}`}
+          ></label>
+        );
+      })}
+    </>
+  );
+}
```
Comment (rohanmathur91) [inline src/components/carousel/Carousel.jsx:23]: Move this component in a separate file

File: src/css/common.css [hunk 5]
```diff
@@ -0,0 +1,118 @@
+@import url(https://halloui.netlify.app/halloUi-share.css);
+/* @import url(/css/homepage.css);
+@import url(/css/header.css);
+@import url(/css/footer.css);
+@import url(/css/variable.css);
+@import url(/css/util.css);
+@import url(/pages/productList/productList.css);
+@import url(/pages/wishlist/wishlist.css); */
```
Comment (rohanmathur91) [inline src/css/common.css:8]: Remove this if not needed

File: src/pages/home/Home.jsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    imgAlt: "bedroom decor",
+    link: "/productList",
+  },
+  {
+    category: "Furnishings",
+    imgSrc: "/assets/images/decor.webp",
+    imgAlt: "hanging light and chair",
+    link: "/productList",
+  },
+];
```
Comment (rohanmathur91) [inline src/pages/home/Home.jsx]: Put this in separate file, then import and use it

File: src/pages/productList/ProductList.jsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+  useEffect(() => {
+    (async function () {
+      try {
+        let products = await axios.get("/api/products");
+        console.log(products);
+        setProductList(products.data.products);
+      } catch (err) {
+        console.log(err);
+      }
```
Comment (rohanmathur91) [inline src/pages/productList/ProductList.jsx]: Remove console log, show the error in UI.

**Review by rohanmathur91 — APPROVED** [review]
Added a few comments.

- Liked that you are using CSS modules
- Also you are looping through the data and not copy-pasting it again which is nice
- Wherever possible you have used reusable components which is also nice.

File: src/components/carousel/Carousel.jsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+      {inputNum.map((item) => {
+        return (
+          <label
+            key={uuid()}
+            htmlFor={item}
+            onClick={() => {
+              setActiveSlide(item);
+            }}
+            className={`${carousel["manual-btn"]}  
+              ${item === activeSlide ? carousel["active"] : "abcd"}`}
```
Comment (Aditi0505) [inline src/components/carousel/Carousel.jsx:17]: Question - what is the significance of "abcd" in the else case?

File: src/components/carousel/carousel.module.css [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+.slides {
+  width: 400%;
+  height: 100vh;
+  display: flex;
+}
+
+/* .slides input {
+  display: none;
+} */
```
Comment (Aditi0505) [inline src/components/carousel/carousel.module.css]: Can remove commented code if not needed.

File: src/components/carousel/carousel.module.css [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  transition: 1s;
+}
+
+.manual-btn:hover {
+  background: var(--primary-color);
+}
+/* 
+#radio1:checked ~ .first {
+  margin-left: 0;
+} */
```
Comment (Aditi0505) [inline src/components/carousel/carousel.module.css]: Here as well. You can remove commented code.

File: src/components/navbar/navbar.module.css [hunk 11]
```diff
@@ -0,0 +1,48 @@
+* {
+  box-sizing: border-box;
+}
+
+.logo {
+  text-decoration: none;
+  color: var(--color-blk);
+  font-size: var(--fs-m);
+  font-family: "Dancing Script", cursive;
+  /* border: 4px double var(--primary-color); */
```
Comment (Aditi0505) [inline src/components/navbar/navbar.module.css]: Can remove commented code here.

**Review by Aditi0505 — APPROVED** [review]
Left few suggestions.
1.The code is really clean.
2. Good use of CSS modules.
3. Good use of  reusable components.

File: src/backend/db/products.js (hunk 2 above)
Comment (surbhi84) [reply src/backend/db/products.js:115]: this is intentional, i'hv left a template for future so i can add more products.

File: src/pages/productList/ProductList.jsx (hunk 7 above)
Comment (surbhi84) [reply src/pages/productList/ProductList.jsx]: done

File: src/components/carousel/Carousel.jsx (hunk 8 above)
Comment (surbhi84) [reply src/components/carousel/Carousel.jsx:17]: left it by mistake, will remove

File: src/components/carousel/Carousel.jsx (hunk 4 above)
Comment (surbhi84) [reply src/components/carousel/Carousel.jsx:23]: this is a small component that is not used anywhere else throughout my application thats why i kept it in the same file for ease of understanding.

File: src/pages/home/Home.jsx (hunk 6 above)
Comment (surbhi84) [reply src/pages/home/Home.jsx]: this is static data, it wont grow any further...i kept it in the same file for ease, let me know if i still should do it and why.

**Comment by surbhi84** [comment]
I'hv made the necessary changes according to reviews, will merge now.

**Pull request merged**